src/Ng/ElfieFromCatalogCollector.cs

```diff
@@ -122,60 +133,33 @@ void ProcessCatalogItems(IEnumerable<JObject> catalogItems)
         /// Process an individual catalog item (NuGet pacakge) which has been added or updated in the catalog
         /// </summary>
         /// <param name="catalogItem">The catalog item to process.</param>
-        void ProcessPackageDetails(JObject catalogItem)
+        async Task ProcessPackageDetailsAsync(CatalogItem catalogItem, CancellationToken cancellationToken)
         {
-            if (IsListed(catalogItem))
-            {
-                Trace.TraceInformation("Processing listed package " + catalogItem["@id"].Value<string>());
-            }
-        }
+            Trace.TraceInformation("#StartActivity ProcessPackageDetailsAsync " + catalogItem.Id + " " + catalogItem.PackageVersion);
 
-        /// <summary>
-        /// Process an individual catalog item (NuGet pacakge) which has been deleted from the catalog
-        /// </summary>
-        /// <param name="catalogItem">The catalog item to process.</param>
-        void ProcessPackageDelete(JObject catalogItem)
-        {
-            Trace.TraceInformation("Processing deleted package " + catalogItem["@id"].Value<string>());
-        }
+            // Download the registration json file.  
+            // The registration file will tell if the package is listed as-well-as provide the package download URL.  
+            RegistrationItem registrationItem = catalogItem.GetRegistrationItem(this._nugetServiceUrls);
 
-        /// <summary>
-        /// Replaces the catalog entry id with the originalId, if it exists.
-        /// </summary>
-        void NormalizeId(JObject catalogItem)
-        {
-            JToken originalId = catalogItem["originalId"];
-            if (originalId != null)
+            // We only need to process listed packages  
+            if (registrationItem.Listed)
             {
-                catalogItem["id"] = originalId.ToString();
+                // TODO: Process the packages here.
             }
-        }
 
-        /// <summary>
-        /// Gets the context node of a catalog entry
-        /// </summary>
-        JToken GetContext(JObject catalogItem)
-        {
-            return catalogItem["@context"];
+            Trace.TraceInformation("#StopActivity ProcessPackageDetailsAsync");
+
         }
 
         /// <summary>
-        /// Determines if a catalog entry is listed based on its published data.
+        /// Process an individual catalog item (NuGet pacakge) which has been deleted from the catalog
         /// </summary>
-        /// <param name="catalogItem"></param>
-        /// <returns>True if the catalog entry is listed. Otherwise false.</returns>
-        bool IsListed(JObject catalogItem)
+        /// <param name="catalogItem">The catalog item to process.</param>
+        async Task ProcessPackageDeleteAsync(CatalogItem catalogItem, CancellationToken cancellationToken)
```

Comment:
I was thinking we would maintain a file which contains the latest stable version for all packages. We would use this file to create the ARDB index (the ARDB index only contains types from the latest stable version.)
When the latest stable version of a package is deleted (delisted) we would need to update this file to remove its entry.